eureka/S5_lightcurve_fitting/models/SinusoidPhaseCurve.py

```diff
@@ -0,0 +1,122 @@
+import numpy as np
+try:
+    import batman
+except ImportError:
+    print("Could not import batman. Functionality may be limited.")
+
+from .Model import Model
+from ..parameters import Parameters
+
+class SinusoidPhaseCurveModel(Model):
+    """A sinusoidal phase curve model"""
+    def __init__(self, transit_model=None, eclipse_model=None, **kwargs):
+        """Initialize the phase curve model
+        """
+        # Inherit from Model calss
+        super().__init__(**kwargs)
+
+        # Define model type (physical, systematic, other)
+        self.modeltype = 'physical'
+
+        # Check for Parameters instance
+        self.parameters = kwargs.get('parameters')
+
+        # Generate parameters from kwargs if necessary
+        if self.parameters is None:
+            self.parameters = Parameters(**kwargs)
+        
+        # Set parameters for multi-channel fits
+        self.longparamlist = kwargs.get('longparamlist')
+        self.nchan = kwargs.get('nchan')
+        self.paramtitles = kwargs.get('paramtitles')
+
+        self.components = None
+        self.transit_model = transit_model
+        self.eclipse_model = eclipse_model
+        if transit_model is not None:
+            self.components = [self.transit_model,]
+        if eclipse_model is not None:
+            if self.components is None:
+                self.components = [self.eclipse_model,]
+            else:
+                self.components.append(self.eclipse_model)
+
+    @property
+    def time(self):
+        """A getter for the time"""
+        return self._time
+
+    @time.setter
+    def time(self, time_array):
+        self._time = time_array
+        if self.transit_model is not None:
+            self.transit_model.time = time_array
+        if self.eclipse_model is not None:
+            self.eclipse_model.time = time_array
+
+    def update(self, newparams, names, **kwargs):
+        super().update(newparams, names, **kwargs)
+        if self.transit_model is not None:
+            self.transit_model.update(newparams, names, **kwargs)
+        if self.eclipse_model is not None:
+            self.eclipse_model.update(newparams, names, **kwargs)
+        return
+
+    def eval(self, **kwargs):
+        """Evaluate the function with the given values"""
+        # Get the time
+        if self.time is None:
+            self.time = kwargs.get('time')
+        
+       #Initialize model
+        bm_params = batman.TransitParams()
+        pc_params = {'A':0., 'B':0., 'C':0., 'D':0.}
```

Comment:
I still think we need better variable names for the PC model. These letters don't convey "PC amplitude" when I see them.

Reply:
AmpCos1, AmpSin1, AmpCos2, AmpSin2?

Reply:
Sure, that works!